code/clean/clean_hi.do

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+
+Clean Hawaii cases
+
+*******************************************************************************/
+
+import excel "$raw_data/HI/hi_raw_cases.xls", sheet("HCRC Closures") firstrow case(lower) clear
+
+
+/*******************************************************************************
+Clean vars
+*******************************************************************************/
+
+// drop random vars 
+drop island eeocno adverseact
+
+// Rename vars
+ren casetype juris
+ren docket id
+ren complaintfiled charge_file_date
+ren basis basis_raw
+ren closed charge_res_date
+ren closurecode outcome
+ren compensation relief
+
+
+/*******************************************************************************
+Clean outcomes
+*******************************************************************************/
+
+g state = "HI"
+
+// Duration
+replace charge_res_date = enfclosure if enfclosure != .
+drop enfclosure
+format charge_file_date %td
+format charge_res_date %td
+g duration = charge_res_date - charge_file_date
+
+// Jurisdiction
+replace juris = "Employment" if juris == "FE"
+replace juris = "Public Accommodation" if juris == "PA"
+replace juris = "Housing" if juris == "RE"
+
+// Basis
+split basis_raw, parse()
+g basis = "Sex" 				if inlist(basis_raw1, "BRE", "DSV", "SEX", "SEX-H", "SEX-P")
+replace basis = "LGBTQ"         if inlist(basis_raw1, "SEX-I", "SOR")
+replace basis = "Religion" 	    if basis_raw1 == "REL"
+replace basis = "Race"		    if basis_raw1 == "RAC"
+replace basis = "Nationality"   if basis_raw1 == "NOR"
+replace basis = "Disability"    if strpos(basis_raw1, "DIS") > 0
+replace basis = "Age" 		    if basis_raw1 == "AGE"
+replace basis = "Retaliation"   if basis_raw1 == "RET"
+replace basis = "Other" 		if inlist(basis_raw1, "ANC", "ARR", "CHI", "COL", "CRE", "FAM", "MAR", "NAT")
+
+// SH
+g sh = 0
+replace sh = 1 if basis_raw1 == "SEX-H"
+drop basis_raw?
```

Comment:
Keep it!

Reply:
The question mark in the back is shorthand and lets Stata know to drop all of basis_raw1, basis_raw2, etc. which were generated when the string was parsed.